okhttp/third_party/okhttp/main/java/io/grpc/okhttp/internal/OkHostnameVerifier.java

```diff
@@ -63,6 +64,9 @@ private OkHostnameVerifier() {
 
   @Override
   public boolean verify(String host, SSLSession session) {
```

Comment:
Let's make `boolean verify(String host, X509Certificate certificate)` private as well (the next method). At that point we know `verifyHostName()` won't receive non-ascii; the ascii-specific case conversations wouldn't technically be needed but we'll keep them to reduce the severity of bugs.